### PR TITLE
move proxy specific logic into proxyutil, implement ReverseProxyClien…

### PR DESCRIPTION
**Pull request by mediocregopher**
…t instead of HTTPProxy

### CODE REVIEW

File: proxyutil/brw.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	}
+}
+
+type brwMarshalled struct {
+	Code   int
+	Header map[string][]string
+	Body   []byte
+}
+
+var _ = func() bool {
```
Comment (jameshartig) [inline proxyutil/brw.go]: why not just do an `init`?

File: proxyutil/proxy.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	AddXForwardedFor(outreq, req)
+
+	res, err := cl.Do(outreq)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, h := range hopHeaders {
```
Comment (jameshartig) [inline proxyutil/proxy.go:90]: Why are you doing this after making the request?

File: proxyutil/brw.go (hunk 1 above)
Comment (mediocregopher) [inline proxyutil/brw.go]: i guess now I can. When this was in rpcutil it was just kinda weird to have it so "far away" in the code

File: proxyutil/proxy.go (hunk 2 above)
Comment (mediocregopher) [inline proxyutil/proxy.go:90]: It's deleting the headers off the response, not the request. It's what httputil did, I didn't really ask questions >_>

**Pull request merged**